userView/controller/user.js

```diff
@@ -0,0 +1,52 @@
+const User = require('../models/User');
+const { StatusCodes } = require('http-status-codes');
+const { BadRequestError, NotFoundError, UnAuthenticatedError } = require('../errors/index');
+const checkPermissions = require('../utils/checkPermissions');
+
+
+const viewCurrentUser = async (req, res) => {
+  res.status(StatusCodes.OK).json({ user: req.user });
+};
+
+// update user with user.save()
+const updateUser = async (req, res) => {
+  const { email, name, lastName, location} = req.body;
+  if (!email || !name || !lastName || !location) {
+    throw new BadRequestError('Please provide all values');
+  }
+  const user = await User.findOne({ _id: req.user.userId });
```

Comment:
Why are you calling a User model here??...I expect a user repository as we already talked about this?